api/controllers/DashboardController.js

```diff
@@ -16,275 +19,400 @@ var dashboardController = {
-                                paginationMessage = 'Showing ' + (offset + 1) + ' &ndash; ' + pageUpperLimit + ' of ' + resultCount + ' applications submitted in the last 60 days';
-                            }
+    _userFriendlyStatuses(casebookStatus, applicationtype) {
+        const eAppStatuses = {
+            Done: {
+                text: 'Completed',
+                colorClass: '', // dark blue
+            },
+            default: {
+                text: 'In progress',
+                colorClass: 'govuk-tag--blue',
+            },
+        };
+
+        const standardStatuses = {
+            Despatched: {
+                text: 'Despatched',
+                colorClass: '', // dark blue
+            },
+            default: {
+                text: casebookStatus,
+                colorClass: 'govuk-tag--blue',
+            },
+        };
+
+        if (!casebookStatus) {
+            return {
+                text: 'Not avialable',
+                colorClass: 'govuk-tag--grey',
+            };
+        }
 
+        if (applicationtype === 'e-Apostille') {
+            return eAppStatuses[casebookStatus] || eAppStatuses['default'];
+        }
+        return standardStatuses[casebookStatus] || standardStatuses['default'];
+    },
 
-                            var view ='dashboard.ejs';
-                            var attributes = {
-                                message: req.flash('info'),
-                                users_applications: results,
-                                moment: moment,
-                                offset: offset,
-                                sortOrder: sortOrder,
-                                paginationMessage: paginationMessage,
-                                currentPage: currentPage,
-                                totalPages: totalPages,
-                                searchCriteria: searchCriteria,
-                                user_data: HelperService.getUserData(req, res),
-                                application_total : c
-                            };
-                            if(req.query.ajax){
-                                view = 'partials/dashboardResults.ejs';
-                                attributes.layout= null;
-                            }
+    _redirectToPage(pageAttributes, req, res) {
+        const { electronicEnabled } = pageAttributes.user_data.user;
+        let view = electronicEnabled
+            ? 'eApostilles/dashboard.ejs'
+            : 'dashboard.ejs';
+
+        if (req.query.ajax) {
+            view = electronicEnabled
+                ? 'partials/dashboardResultsEApp.ejs'
+                : 'partials/dashboardResults.ejs';
+            pageAttributes.layout = null;
+        }
 
+        return res.view(view, pageAttributes);
+    },
 
-                            return res.view(view,attributes);
-                        }
-                    );
+    _paginationAndPageTotal(results, offset, pageSize) {
```

Comment:
The pagination is incorrect when showing a mixture of e-Apps and regular applications.

![Screenshot 2021-10-04 at 15 21 35](https://user-images.githubusercontent.com/1797978/135868543-a2f8b398-b027-4ecf-8058-beecd9529e6d.png)

Reply:
This is super odd. I am able to get the pagination working locally on my machine with a mixture of both app types

https://user-images.githubusercontent.com/1377253/135899470-0008dd59-883f-47f1-a61c-a8f038abf7d9.mp4

Would you like to have a chat about this tomorrow? Maybe there's a chance a stored function/procedure hasn't been installed or something 🤔

Reply:
Hey Simon this has been fixed now. Man, I dislike stored procedures. If you reapply the updated function in EIA-101_update_dashboard_function.sql it should work 🤞